packages/muze-utils/src/store.js

```diff
@@ -0,0 +1,578 @@
+        }
+        return this;
+    }
+
+    /**
+     * This method is used to update the value of a property in the state store.
+     *
+     * @param {string} propName The name of the property.
+     * @param {number} value The new value of the property.
+     * @memberof Store
+     */
+    commit (propName, value, namespace, disableListener = false) {
+        const commits = this._commits;
+        const savedCommits = this._savedCommits;
+        const locked = getObjProp(commits, propName, 'locked');
+        const propListenerMap = this._propListenerMap[propName] || {};
+
+        propListenerMap.disabled = disableListener;
+
+        if (locked) {
+            commits[propName].queue.push([propName, value, namespace]);
+            return this;
+        }
+
+        let sanitizedVal = value;
+        if (namespace) {
+            if (this._locked) {
+                initProp(this._queuedProps, [propName]);
+                this._queuedProps[propName][namespace] = value;
+                sanitizedVal = this._queuedProps[propName];
+            } else {
+                sanitizedVal = defaultValue(this.get(propName), {});
+                sanitizedVal[namespace] = value;
+            }
+
+            types.forEach((type) => {
+                initProp(savedCommits, [propName, type]);
+                savedCommits[propName][type][namespace] = true;
+                if (getObjProp(propListenerMap, type, namespace)) {
+                    propListenerMap[type][namespace].fns = propListenerMap[type][namespace]._fnCount;
+                }
+            });
+        }
+
+        this.model.prop(propName, sanitizedVal);
+        return this;
+    }
+
+    /**
+     * This method is used to register a callbacl that will execute
+     * when one or more properties change.
+     *
+     * @param {string | Array} propNames name of property or array of props.
+     * @param {Function} callBack The callback to execute.
+     * @memberof Store
+     */
+    /* istanbul ignore next */registerChangeListener (propNames, callBack, instantCall, namespaceInf = {}) {
+        let props = propNames;
+        if (!Array.isArray(propNames)) {
+            props = [propNames];
+        }
+        registerListener(this, 'next', {
+            namespaceInf,
+            callBack,
+            instantCall,
+            props
+        });
+        return this;
+    }
+    /**
+     * This method is used to register a callbacl that will execute
+     * when one or more properties change.
+     *
+     * @param {string | Array} propNames name of property or array of props.
+     * @param {Function} callBack The callback to execute.
+     * @memberof Store
+     */
+    /* istanbul ignore next */ registerImmediateListener (propNames, callBack, instantCall, namespaceInf = {}) {
+        let props = propNames;
+        if (!Array.isArray(propNames)) {
+            props = [propNames];
+        }
+        registerListener(this, 'on', {
+            namespaceInf,
```

Comment:
Immediate and change listener appear to be the exact same, barring one word. Use single function and pass appropriate params.